controllers/crypto/productsController.js

```diff
@@ -0,0 +1,79 @@
+const logger = require('../../utils/logger')
+const productQuery = require('../../models/crypto/products')
+
+/**
+ * Fetches the data of products
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getProducts = async (req, res) => {
+  try {
+    const allProducts = await productQuery.fetchProducts()
+
+    return res.json({
+      message: 'Products returned successfully!',
+      Products: allProducts
+    })
+  } catch (error) {
+    logger.error(`Error while fetching all products: ${error}`)
+    return res.boom.serverUnavailable('Something went wrong please contact admin')
+  }
+}
+
+/**
+ * Fetches the data od produc
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getProduct = async (req, res) => {
+  try {
+    const product = await productQuery.fetchProduct(req.params.id)
+    if (product) {
+      return res.status(200).json({
+        message: 'Products returned successfully!',
+        Products: product
+      })
+    }
+    return res.status(404).json({
+      message: 'Products not found!'
+    })
```

Comment:
We are using boom for errors,
We can write `return res.boom.notFound('<Error Message>')`

Boom docs -
https://npmdoc.github.io/node-npmdoc-boom/build/apidoc.html